docs/source/nlp/machine_translation.rst

```diff
@@ -0,0 +1,251 @@
+
+2. Bucketing - Sentences vary in length and when creating minibatches, we'd like sentences in them have roughly the same length to minimize the number of <pad> tokens, to maximize computational efficiency. This step groups sentences on roughly the same length into buckets.
+
+3. Batching and padding - Creates minibatches of with a maximum number of tokens specified by `model.{train_ds,validation_ds,test_ds}.tokens_in_batch` from buckets and pads the sequences to pack them so they can be packed into a tensor.
+
+Datasets can be configured as follows
+
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| **Parameter**                                               | **Data Type**   |   **Default**  | **Description**                                                                                |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.src_file_name        | str             | null           | Path to the source language file                                                               |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.tgt_file_name        | str             | null           | Path to the target language file                                                               |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.tokens_in_batch      | int             | 512            | Maximum number of tokens per minibatch                                                         |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.clean                | bool            | true           | Whether to clean the dataset by discarding examples that are greater than max_seq_length       |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.max_seq_length       | int             | 512            | Path to pre-computed vocab file if exists                                                      |
```

Comment:
Description is for vocab_file should be for max_seq_length

Reply:
Done. Thanks!